pkg/combination/combination_test.go

```diff
@@ -35,25 +35,46 @@ var combinationTests = []struct {
 			err:          nil,
 		},
 	},
+	{
+		name:  "standard set of long args",
+		input: testdata.LongCombinationInput,
+		expected: expected{
+			combinations: testdata.LongCombinationOutput,
+			err:          nil,
+		},
+	},
 }
 
-func TestAll(t *testing.T) {
+func TestNext(t *testing.T) {
+	// Test 1: Uses testIterative()
 	for _, tt := range combinationTests {
 		t.Run(tt.name, func(t *testing.T) {
 			combinationStream := NewStream(
 				WithArgs(tt.input),
-				WithSolveAhead(),
 			)
-			got, err := combinationStream.All()
-			if !errors.Is(err, tt.expected.err) {
-				t.Fatal("error received while retreiving all combinations:", err)
+
+			var got []map[string]string
```

Comment:
nit/not blocking: we may want to refactor this test packages setup to use [table-driven testing](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) and add the `got` and a `want` fields to a custom struct. It looks like we also check the types of errors that are produced, so having an expectErr, or wantErr field could be appropriate too.

Reply:
Yeah, I agree. It made more sense when we were using the same data for two tests but does not anymore. A good example of what this should look like now can be found in the builder test.

https://github.com/operator-framework/combo/blob/main/pkg/template/builder_test.go

Reply:
Good catch - I meant to link to a concrete example, but it looks like that fell through the cracks. Try and take a look at what Tyler had linked as it should paint a decent picture of the setup we're talking about.